### PR TITLE
Add a FAQ.

### LINKED ISSUE #621: Add a FAQ
I've been going through all issues posted to this bug tracker over the years to build a list of questions that are actually frequently asked.

Here's the result.

**websockets issues**

- [x] send / receive more than one message #83 #122 #148 #152 #398 #548 #559 #564 #591
- ~~debugging connection establishment => logging.DEBUG~~
- ~~network errors in logs ... #448~~

**asyncio issues**

- [x] only one client can connect at a time #136 #187 #206 #584 #620
- ~~forgot to await a coroutine ... #561~~
- [x] setup, fix or disable SSL verification ... #419 #480 #522 #610

**websockets design**

- [x] high level patterns: broadcast / subscriptions / channels / rooms #17 #101 #384 #427 #481 #603
- [x] use callbacks instead of coroutines #32 #52 #103 #161 #173 #243 #344
- [x] interact synchronously with the API #485
- [x] WS and HTTP on the same port #49 #116
- [x] Python 2 support #125 #191

**How do I... (general, websockets)**

- [x] respond to pings #79 #268 #473 #532
- [x] keep connections alive (now built-in and enabled by default) #110 #333 #446
- [x] reconnect automatically #414 #367
- [x] close individual client connection, server connection, server gracefully? #362 #444 #457
- [x] set a timeout #141

**How do I... (asyncio)**

- [x] do two things at once #37 #45 #48 #100 #247 #382 #490 #526
- ~~attempts to use multiple threading #343 #346~~
- ~~run event loop in another thread #360~~

**How do I... (specific, websockets)**

- [x] get the remote IP address #33 #66 #555
- [x] bind to a given port on the client or server side ... #602
- [x] access HTTP headers #82 #491
- [x] sending HTTP headers #420

### PR DESCRIPTION
**Pull request by aaugustin**
Fix #621.

### CODE REVIEW

**Comment by aaugustin** [comment]
If you're reading this and have 3 minutes to read my proposal, I'd be very happy to get feedback before merging it :-)

File: docs/faq.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+:attr:`~server.WebSocketServerProtocol.process_request` hook.
+If you need more, pick a HTTP server and run it separately.
+
+Client side
+-----------
+
+How do I close a connection properly?
+.....................................
+
+Use :func:`connect` as a context manager and trust websockets::
```
Comment (lgrahl) [inline docs/faq.rst]: Is that sentence incomplete? *Trust websockets* regarding?

File: docs/faq.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+websockets provides high-level, coroutine-based APIs. Compared to callbacks,
+coroutines make it easier to manage control flow in concurrent code.
+
+If you prefer callback-based APIs, you should use another library.
+
+Can I use ``websockets`` synchronously, without ``async`` / ``await``?
+......................................................................
+
+You can convert every asynchronous call to a synchronous call by wrapping it
+in ``asyncio.get_event_loop().run_until_complete(...)``.
```
Comment (lgrahl) [inline docs/faq.rst]: I think there is `asyncio.run` now.

File: docs/faq.rst (hunk 1 above)
Comment (aaugustin) [reply docs/faq.rst]: Implicitly, "to do the right thing". I'll phrase it differently.

File: docs/faq.rst (hunk 2 above)
Comment (aaugustin) [reply docs/faq.rst]: It won't work here because it always creates a new event loop and closes it at the end.

**Pull request merged**